### PR TITLE
Add aws dms support

### LINKED ISSUE #145: Add support for AWS Database Migration Service (DMS)
https://aws.amazon.com/dms/

https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/dms_replication_instance

### PR DESCRIPTION
**Pull request by sinabakh**
Fix #145 

AWS Pricing: https://aws.amazon.com/dms/pricing/

Terraform: https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/dms_replication_instance

### CODE REVIEW

**Comment by sinabakh** [comment]
Tests failure is related to #148 so I think we can merge this PR with no problem.

**Comment by sinabakh** [comment]
I didn't add anything to examples, is there any need to do it?

File: internal/providers/terraform/aws/dms.go [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+	case "t3":
+		freeStorageSize = 50
+	}
+	var storageSize int64 = 0
+	if baseStorageSize > freeStorageSize {
+		storageSize = baseStorageSize - freeStorageSize
+	}
+
+	return &schema.CostComponent{
+		Name:            fmt.Sprintf("General Purpose Storage (SSD - %s family)", instanceFamily),
```
Comment (alikhajeh1) [inline internal/providers/terraform/aws/dms.go:81]: General Purpose Storage -> General purpose storage

Comment (alikhajeh1) [inline internal/providers/terraform/aws/dms.go:81]:
```suggestion
		Name:            fmt.Sprintf("General purpose storage (gp2)"),
```

File: internal/providers/terraform/aws/dms.go [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+		Unit:            "hours",
```
Comment (alikhajeh1) [inline internal/providers/terraform/aws/dms.go:82]: won't this be GB-months?

**Review by alikhajeh1 — APPROVED** [review]
Thanks! LGTM.
To keep the cost component names consistent, I suggest using the same naming scheme for storage as https://docs.google.com/spreadsheets/d/1H_bn2jLzYr7xyrvNsFn-0rDaGPGpnrVTPsjVHzr-kM4/edit#gid=0 since the instanceType itself doesn't impact the storage cost.
https://calculator.aws/#/createCalculator shows a similar thing (gp2).

**Comment by alikhajeh1** [comment]
> I didn't add anything to examples, is there any need to do it?

Nope. We want to keep the example short as it is.

**Comment by alikhajeh1** [comment]
> Tests failure is related to #148 so I think we can merge this PR with no problem.

yep, this is fixed in master.

File: internal/providers/terraform/aws/dms.go (hunk 1 above)
Comment (sinabakh) [reply internal/providers/terraform/aws/dms.go]: Fixed

File: internal/providers/terraform/aws/dms.go (hunk 2 above)
Comment (sinabakh) [reply internal/providers/terraform/aws/dms.go]: Fixed

**Comment by sinabakh** [comment]
@alikhajeh1 Thanks for the detailed code-review. All comments are resolved now.

**Pull request merged**